orttraining/orttraining/python/experimental/amp/loss_scaler.py

```diff
@@ -27,12 +29,12 @@ def loss_scale(self):
 
     @loss_scale.setter
     def loss_scale(self, loss_scale):
-        assert isinstance(loss_scale, float) and loss_scale > 0, "'loss_scale' must be a positive float"
-        self._loss_scale = loss_scale
+        assert isinstance(loss_scale, (int, float)) and loss_scale > 0, "'loss_scale' must be a positive float"
+        self._loss_scale = float(loss_scale)
 
     def reset(self):
         r"""Resets loss scaler internal state"""
-        raise NotImplementedError
+        self._loss_scale = float(self._initial_loss_scale)
```

Comment:
nit: casting seems unnecessary

Reply:
you are right. it was added as an extra protection in case the user sneaks into the class and manually change it